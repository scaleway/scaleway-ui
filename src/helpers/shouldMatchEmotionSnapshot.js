/* eslint-disable import/no-extraneous-dependencies */
import createCache from '@emotion/cache'
import { createSerializer } from '@emotion/jest'
import { CacheProvider, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import React from 'react'
import theme from '../theme'

expect.addSnapshotSerializer(createSerializer())

const emotionCache = createCache({
  key: 'cache',
})
emotionCache.compat = true

export default async (component, { transform } = {}) => {
  const node = render(
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </CacheProvider>,
  )
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
}
