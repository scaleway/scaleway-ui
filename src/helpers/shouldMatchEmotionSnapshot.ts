/* eslint-disable import/no-extraneous-dependencies */
import { createSerializer } from '@emotion/jest'
import { RenderOptions, render } from '@testing-library/react'
import { ReactNode } from 'react'
import defaultTheme from '../theme'
import renderWithTheme from './renderWithTheme'

// use only class hash (generated from css style content)
expect.addSnapshotSerializer(
  createSerializer({ classNameReplacer: className => className }),
)

interface Options {
  options?: RenderOptions
  theme?: typeof defaultTheme
  transform?: (node: ReturnType<typeof render>) => Promise<void> | void
}

export default async (
  component: ReactNode,
  { transform, options, theme }: Options = {},
): Promise<void> => {
  const node = renderWithTheme(component, options, theme)
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
  node.unmount()
}
