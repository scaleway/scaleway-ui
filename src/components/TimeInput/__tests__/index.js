import React from 'react'
import { TimeInput } from '..'
import shouldMatchEmotionSnapshot from '../../../helpers/shouldMatchEmotionSnapshot'

describe('TimeInput', () => {
  test('renders correctly with base props', () => {
    shouldMatchEmotionSnapshot(
      <TimeInput inputId="test" labelId="test-label" name="timeinput-test-0" />,
    )
  })
})
