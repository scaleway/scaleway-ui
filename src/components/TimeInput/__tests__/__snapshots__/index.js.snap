// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeInput renders correctly with base props 1`] = `
<DocumentFragment>
  .emotion-1 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
}

.emotion-2 {
  z-index: 9999;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  padding: 0;
  white-space: nowrap;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border-color: #dce1eb;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  box-shadow: none;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 48px;
  outline: 0!important;
  position: relative;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  box-sizing: border-box;
  color: #4a4f62;
  font-size: 16px;
  font-weight: 500;
  line-height: 24px;
  -webkit-animation: none;
  animation: none;
}

.emotion-3:hover {
  border-color: hsl(0, 0%, 70%);
}

.emotion-3:focus-within {
  border-color: #4f0599;
  box-shadow: 0 0 2px 2px rgba(79,5,153,0.25);
}

.emotion-3:focus-within svg {
  fill: #4f0599;
}

.emotion-3:hover {
  border-color: #4f0599;
}

.emotion-3:hover svg {
  fill: #4f0599;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 8px;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
  box-sizing: border-box;
  height: 100%;
}

.emotion-6 {
  position: absolute;
  left: 0;
  font-weight: 400;
  pointer-events: none;
  color: #b2b6c3;
  white-space: nowrap;
  width: 100%;
  height: 100%;
  font-size: 16px;
  -webkit-transition: -webkit-transform 250ms ease;
  transition: transform 250ms ease;
  opacity: 0;
  -webkit-transform: translate(0, -8px) scale(0.8);
  -moz-transform: translate(0, -8px) scale(0.8);
  -ms-transform: translate(0, -8px) scale(0.8);
  transform: translate(0, -8px) scale(0.8);
  transform-origin: left;
  padding-left: 8px;
  left: 0;
  top: 2px;
  opacity: 1;
}

.emotion-7 {
  color: #4a4f62;
  margin-left: 0;
  margin-right: 0;
  max-width: calc(100% - 8px);
  overflow: hidden;
  position: absolute;
  text-overflow: ellipsis;
  white-space: nowrap;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  box-sizing: border-box;
  margin-top: 5px;
  padding-left: 0;
}

.emotion-8 {
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 0;
  visibility: visible;
  color: hsl(0, 0%, 20%);
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 0;
}

.emotion-9 {
  padding-top: 11px;
  margin-left: 0px;
}

.emotion-10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  max-height: 48px;
}

.emotion-11 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: #eeeeff;
  margin-bottom: 8px;
  margin-top: 8px;
  width: 1px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-12 {
  color: hsl(0, 0%, 80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-12:hover {
  color: hsl(0, 0%, 60%);
}

.emotion-14 {
  fill: #d4dae7;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.emotion-14.emotion-14 {
  vertical-align: middle;
  margin-right: 0;
}

.emotion-15 {
  -webkit-transition: max-height 300ms ease-out,opacity 300ms ease-out;
  transition: max-height 300ms ease-out,opacity 300ms ease-out;
  max-height: 0;
  overflow: hidden;
  opacity: 0;
  height: auto;
  margin-top: 0;
}

.emotion-15.emotion-15 {
  overflow: hidden;
}

.emotion-16.emotion-16 {
  font-size: 12px;
  color: #ef5774;
  padding-top: 2px;
}

<div
    class=" emotion-0 emotion-1"
    data-testid="rich-select-timeinput-test-0"
  >
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="emotion-2"
    />
    <div
      class=" emotion-3"
    >
      <div
        class=" emotion-4"
      >
        <label
          aria-live="assertive"
          class="emotion-5 emotion-6"
          for="test"
          id="test-label"
        >
          Select...
        </label>
        <div
          class=" emotion-7"
        >
          00:00
        </div>
        <div
          class="emotion-8"
        >
          <div
            class="emotion-9"
            style="caret-color: transparent; display: inline-block;"
          >
            <input
              aria-autocomplete="list"
              aria-controls="test-label"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              id="test"
              spellcheck="false"
              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            />
          </div>
        </div>
      </div>
      <div
        class=" emotion-10"
      >
        <span
          class=" emotion-11"
        />
        <div
          aria-hidden="true"
          class=" emotion-12"
        >
          <svg
            class="sc-ui-icon emotion-13 emotion-14"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22C6.47,22 2,17.5 2,12A10,10 0 0,1 12,2M12.5,7V12.25L17,14.92L16.25,16.15L11,13V7H12.5Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="timeinput-test-0"
      type="hidden"
      value="00:00"
    />
    <div
      class="emotion-15"
    >
      <div
        class="emotion-16"
      />
    </div>
  </div>
</DocumentFragment>
`;
